File: samplejavaapp/src/main/java/com/yoox/samplejavaapp/common/PermissionUtils.java
package com.yoox.samplejavaapp.common;

import android.app.Activity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import static android.content.pm.PackageManager.PERMISSION_GRANTED;

public class PermissionUtils {

    public static boolean askPermission(String permission, Activity activity, int requestCode) {
        if (ContextCompat.checkSelfPermission(activity, permission) == PERMISSION_GRANTED) {
            return true;
        } else {
            ActivityCompat.requestPermissions(activity, new String[]{permission}, requestCode);
            return false;
        }
    }
}

File: samplekotlinapp/src/main/java/com/yoox/samplekotlinapp/common/PermissionUtils.kt
package com.yoox.samplekotlinapp.common

import android.app.Activity
import android.content.pm.PackageManager.PERMISSION_GRANTED
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat


fun Activity.askPermission(permission: String, requestCode: Int): Boolean {
    return if (ContextCompat.checkSelfPermission(this, permission) == PERMISSION_GRANTED) {
        true
    } else {
        ActivityCompat.requestPermissions(this, arrayOf(permission), requestCode)
        false
    }
}

